tests/simulation/test_circuit.py

```diff
@@ -51,14 +56,20 @@ def test_alternative_circuit_apply(BackendClass):
 def test_circuit_apply_None_raises_ValueError(CircuitClass):
     with pytest.raises(ValueError):
         circ = CircuitClass(nqubit=2, depth=1, gates=standard_gates)
-        circ.apply(gate=None, i=0)
+        if isinstance(circ, BinaryCircuit):
+            circ.update_circuit_list(gate=None, qubit = [0])
+        else:
+            circ.apply(gate=None, i=0)
 
 
 @pytest.mark.parametrize("CircuitClass", circuit_classes)
 def test_circuit_apply_2qubit_gate_raises_ValueError(CircuitClass):
     with pytest.raises(ValueError):
         circ = CircuitClass(nqubit=2, depth=1, gates=standard_gates)
-        circ.apply(gate=helper_gates.CNOT, i=0) # Use apply for 2 qubit (CNOT) gate.
+        if isinstance(circ, BinaryCircuit):
+            circ.update_circuit_list(gate=helper_gates.CNOT, qubit = [0])
```

Comment:
@paolodr98 It seems that the two circuit classes do not have the same interface. Is it necessary that there is a mismatch? Would it be possible to make the interface match with reasonable effort?

Reply:
Just to ensure compatibility